src/main/java/org/jenkinsci/plugins/runselector/selectors/DisplayNameRunSelector.java

```diff
@@ -0,0 +1,50 @@
+package org.jenkinsci.plugins.runselector.selectors;
+
+import hudson.Extension;
+import hudson.model.Job;
+import hudson.model.Run;
+import org.jenkinsci.Symbol;
+import org.jenkinsci.plugins.runselector.RunSelectorDescriptor;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Selects the build based on its display name.
+ *
+ * @author Alexandru Somai
+ */
+public class DisplayNameRunSelector extends AbstractSpecificRunSelector {
```

Comment:
Agree, it makes sense to make it `DisplayNameRunFilter`, as you have suggested in #14